main.cpp

```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
 paste your code below
 */
 
+/*
+Thing 1) Castle
+5 properties:
+    1) number of gates (int)
+    2) name of king (std::string)
+    3) name of queen (std::string)
+    4) amount of taxes collected (float)
+    5) is at war (bool)
+3 things it can do:
+    1) collect tax from peasants.
+    2) open || close gates.
+    3) give orders to the kingdom.
+*/
+struct Castle
+{
+// 5 properties:
+    int numberOfGates = 8;
+    std::string nameOfKing = "Edward";
+    std::string nameOfQueen = "Magda";
+    float amountOfTaxesCollected = 349.77f;
+    bool isAtWar = false;
+
+    // NESTED UDT
+    struct FarmlandRevenue
+    {
+        int farmlandArces = 65;
+        int numberOfWorkers = 120;
+        float wheatSalePricePerKG = 1.8f;
+        std::string farmlandType = "Cereals and Grains";
+        bool isFarmlandInDraught = false;
+
+        void sellWheat(int wheatStockLevel, float currentMarketPrice, int amountToSell);
+
+        float displayCurrentTurnover(int totalWorkers, int farmlandArea, float currentMarketPrice, float previousMarketPrice); // NOTE: returns a value that would match the estimated turnover total.
+
+        void floodFarmlands(int farmlandToFold, float amountOfFlooding, bool farmlandDraught);
+
+    };
+
+// 3 things it can do:
+    // 1) collect tax from peasants.
+    float reportCollectTaxes(float collectTaxes, FarmlandRevenue farmlandIncome); // NOTE: Parameter required for a value to perform the function.
+
+    // 2) open || close gates.
+    void openCastleGates(int numberOfGatesClosed);
 
+    // 3) give orders to the kingdom.
+    std::string announceOrders(std::string orders); // NOTE: Function returns a string of text and implements the input parameter text.
 
+    FarmlandRevenue agricultureWorth;
+
+};
+
+/*
+Thing 2) MIDI Keyboard
+5 properties:
+    1) number of keys (int)
+    2) number of knobs (int)
+    3) number of MPC pads (int)
+    4) pitch control range (double)
+    5) mode selection (char)
+3 things it can do:
+    1) adjust pitch.
+    2) register key presses.
+    3) register pad hit velocities.
+*/
+struct MidiKeyboard
+{
+// 5 properties:
+    int numberOfKeys = 49;
+    int numberOfKnobs = 12;
+    int numberOfMPCPads = 8;
+    double pitchControlRange = 200000.0;
+    char modeSelection = 'X';
+};
+// 3 things it can do:
+//     1) adjust pitch.
+    double adjustPitch(MidiKeyboard pitchChange); // NOTE: the pitch control effects the adjustment of the functions return value.
+//     2) register key presses.
+    void isKeyPressed();
 
+//     3) register pad hit velocities.
+    float padHitAmount(MidiKeyboard padNumber); // NOTE: returns a relevant value able to influence other fuctions.
```

Comment:
These have not been declared as member functions
they are outside the struct's closing `}`
- [ ] revise